src/PhpParser/Node/Manipulator/PropertyManipulator.php

```diff
@@ -32,6 +33,11 @@ final class PropertyManipulator
 {
     use DoctrineTrait;
 
+    /**
+     * @var string[]
+     */
+    private const MODIFIED_FUNCTION_NAMES = ['end'];
```

Comment:
@Aerendir Could you think of other functions that change passed argument?

Reply:
I'm not sure I'm understanding what you mean by "change passed argument".

But, if the example is `end()` function, then there are a lot of functions related to arrays that change the passed argument.

The ones that comes to my mind:

- https://www.php.net/manual/en/function.array-change-key-case.php
- https://www.php.net/manual/en/function.array-flip.php
- https://www.php.net/manual/en/function.array-pop.php
...

Do you mean this kind of functions?

Reply:
Yes, exactly

Reply:
Could you send PR to add them after this is merged?

Reply:
For the moment I don't think will find the time... :(

I'm upgrading all my libraries and it is a tedious and long process... 😩

I should send a PR to Mautic and I'm not able to find 10 minutes 😓

Reply:
Sure :+1: 

Whenever you're ready (or the rule fails again :D)